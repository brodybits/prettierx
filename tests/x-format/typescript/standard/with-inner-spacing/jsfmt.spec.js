// [prettierx] test script notice:
// This test script runs for test files in parent directory,
// **not** on any files in *this* directory.

const path = require("path");
const dirpath = path.join(__dirname, "/..");

run_spec(dirpath, ["typescript"], {
  spaceInParens: true,
  yieldStarSpacing: true,
  generatorStarSpacing: true,
  typeAngleBracketSpacing: true,
  spaceBeforeFunctionParen: true,
  singleQuote: true,
  jsxSingleQuote: true,
  semi: false,
});
