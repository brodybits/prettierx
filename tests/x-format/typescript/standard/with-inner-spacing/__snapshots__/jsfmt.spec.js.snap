// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`correct.ts - {"spaceInParens":true,"yieldStarSpacing":true,"generatorStarSpacing":true,"typeAngleBracketSpacing":true,"spaceBeforeFunctionParen":true,"singleQuote":true,"jsxSingleQuote":true,"semi":false} format 1`] = `
====================================options=====================================
generatorStarSpacing: true
jsxSingleQuote: true
parsers: ["typescript"]
printWidth: 80
semi: false
singleQuote: true
spaceBeforeFunctionParen: true
spaceInParens: true
typeAngleBracketSpacing: true
yieldStarSpacing: true
                                                                                | printWidth
=====================================input======================================
type IdentityWrapper = {
  func<T>(T): T
}

const identity2 = <T>(t: T): T => t

const a = 1

function identity<T> (value: T): T {
  return value
}

interface Foo<T> {
  getter(value: T): T
}

class Example<T> implements Foo<T> {
  value: T
  getter (value: T): T {
    return this.value
  }
  set setter (value: T) {
    this.value = value
  }
}

=====================================output=====================================
type IdentityWrapper = {
  func< T >( T ): T
}

const identity2 = < T >( t: T ): T => t

const a = 1

function identity< T > ( value: T ): T {
  return value
}

interface Foo< T > {
  getter( value: T ): T
}

class Example< T > implements Foo< T > {
  value: T
  getter ( value: T ): T {
    return this.value
  }
  set setter ( value: T ) {
    this.value = value
  }
}

================================================================================
`;

exports[`incorrect.ts - {"spaceInParens":true,"yieldStarSpacing":true,"generatorStarSpacing":true,"typeAngleBracketSpacing":true,"spaceBeforeFunctionParen":true,"singleQuote":true,"jsxSingleQuote":true,"semi":false} format 1`] = `
====================================options=====================================
generatorStarSpacing: true
jsxSingleQuote: true
parsers: ["typescript"]
printWidth: 80
semi: false
singleQuote: true
spaceBeforeFunctionParen: true
spaceInParens: true
typeAngleBracketSpacing: true
yieldStarSpacing: true
                                                                                | printWidth
=====================================input======================================
type IdentityWrapper = {
  func <T>(T) :T
}

function identity <T>(value :T) :T {
  return value
}

interface Foo <T>{
  getter(value:T):T
}

class Example<T> implements Foo <T>{
  value:T;
  getter(value :T) :T {
    return this.value
  }
  set setter(value :T) {
    this.value = value
  }
}

=====================================output=====================================
type IdentityWrapper = {
  func< T >( T ): T
}

function identity< T > ( value: T ): T {
  return value
}

interface Foo< T > {
  getter( value: T ): T
}

class Example< T > implements Foo< T > {
  value: T
  getter ( value: T ): T {
    return this.value
  }
  set setter ( value: T ) {
    this.value = value
  }
}

================================================================================
`;
